libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java

```diff
@@ -144,19 +146,33 @@ private static String findAgentJar() {
      * @throws IllegalStateException if the entitlements system can't prevent an unauthorized action of our choosing
      */
     private static void selfTest() {
-        ensureCannotStartProcess();
-        ensureCanCreateTempFile();
+        ensureCannotStartProcess(false);
+        ensureCannotStartProcess(true);
+        ensureCanCreateTempFile(false);
+        ensureCanCreateTempFile(true);
     }
 
-    private static void ensureCannotStartProcess() {
+    private static void ensureCannotStartProcess(boolean useReflection) {
```

Comment:
Nit: I'm not a fan of booleans (and ifs 😄) ; I would separate these in 2 functions, but it's up to you.
Otherwise, I would add a comment on why we are using reflection (the same you have in the description of the PR here, or something similar).

(I think that splitting and renaming make the comment superfluous:
```
ensureCannotStartProcess()
ensureCannotStartProcessViaReflection()
```
but again it's a preference, your call)

Reply:
I share your boolean allergy, but that approach would end up duplicating all but one line of code in each method.

Maybe cleaner with lambdas?

Reply:
Did it with method refs. I think it looks pretty good.

Reply:
I agree, it looks good! Thank you!